pytestqt/wait_signal.py

```diff
@@ -54,11 +57,77 @@ def _quit_loop_by_timeout(self):
             self._loop.quit()
 
     def _cleanup(self):
+        # store timeout message before the data to construct it is lost
+        self._timeout_message = self.get_timeout_error_message()
         if self._timer is not None:
             _silent_disconnect(self._timer.timeout, self._quit_loop_by_timeout)
             self._timer.stop()
             self._timer = None
 
+    def get_timeout_error_message(self):
+        """Subclasses have to implement this, returning an appropriate error message for a SignalTimeoutError."""
+        raise NotImplementedError
+
+    def _extract_pyqt_signal_name(self, potential_pyqt_signal):
+        signal_name = potential_pyqt_signal.signal  # type: str
+        if not isinstance(signal_name, str):
+            raise TypeError("Invalid 'signal' attribute in {}. "
+                            "Expected str but got {}".format(signal_name, type(signal_name)))
+        # strip magic number "2" that PyQt prepends to the signal names
+        signal_name = signal_name.lstrip('2')
+        return signal_name
+
+    def _extract_signal_from_signal_tuple(self, potential_signal_tuple):
+        if isinstance(potential_signal_tuple, tuple):
+            if len(potential_signal_tuple) != 2:
+                raise ValueError("Signal tuple must have length of 2 (first element is the signal, "
+                                 "the second element is the signal's name).")
+            signal_tuple = potential_signal_tuple
+            signal_name = signal_tuple[1]
+            if not isinstance(signal_name, str) or not signal_name:
+                raise TypeError("Invalid type for user-provided signal name, "
+                                "expected str but got {}".format(type(signal_name)))
```

Comment:
You're right. I'll fix this, i.e. split it into two ``if``s (each one without any ``or``).